src/mgmt/cli_mgmt.c

```diff
@@ -195,10 +218,501 @@ dc_mgmt_profile(uint64_t modules, char *path, bool start)
+	uint32_t	syb_magic;
+	uint32_t	syb_npsrbs;
+	char		syb_name[DAOS_SYS_NAME_MAX + 1];
+	struct psr_buf	syb_psrbs[0];
+};
+
+/* Convert psrbs[npsrbs] to psrs[npsrs]. */
+static int
+get_attach_info_from_buf(int npsrbs, struct psr_buf *psrbs, int *npsrs,
+			 struct dc_mgmt_psr **psrs)
+{
+	struct dc_mgmt_psr     *p;
+	int			i;
+	int			rc = 0;
+
+	D_ALLOC_ARRAY(p, npsrbs);
+	if (p == NULL)
+		return -DER_NOMEM;
+
+	for (i = 0; i < npsrbs; i++) {
+		p[i].rank = psrbs[i].psrb_rank;
+		D_ASPRINTF(p[i].uri, "%s", psrbs[i].psrb_uri);
+		if (p[i].uri == NULL) {
+			rc = -DER_NOMEM;
+			break;
+		}
+	}
+	if (rc != 0) {
+		for (; i >= 0; i--) {
+			if (p[i].uri != NULL)
+				D_FREE(p[i].uri);
+		}
+		D_FREE(p);
+		return rc;
+	}
+
+	*npsrs = npsrbs;
+	*psrs = p;
+	return 0;
+}
+
+static void
+put_attach_info(int npsrs, struct dc_mgmt_psr *psrs)
+{
+	int i;
+
+	for (i = 0; i < npsrs; i++)
+		D_FREE(psrs[i].uri);
+	D_FREE(psrs);
+}
+
+static int
+attach_group(const char *name, bool pmixless, int npsrs,
+	     struct dc_mgmt_psr *psrs, crt_group_t **groupp)
+{
+	crt_group_t    *group;
+	int		i;
+	int		rc;
+
+	if (!pmixless) {
+		rc = crt_group_attach((char *)name, &group);
+		if (rc != 0) {
+			D_ERROR("failed to attach to group %s: %d\n", name, rc);
+			goto err;
+		}
+		goto out;
+	}
+
+	rc = crt_group_view_create((char *)name, &group);
+	if (rc != 0) {
+		D_ERROR("failed to create group %s: %d\n", name, rc);
+		goto err;
+	}
+
+	for (i = 0; i < npsrs; i++) {
+		crt_node_info_t	info;
+
+		info.uri = psrs[i].uri;
+		rc = crt_group_node_add(group, psrs[i].rank, 0 /* tag */, info);
+		if (rc != 0) {
+			D_ERROR("failed to add rank %u URI %s to group %s: "
+				"%d\n", psrs[i].rank, psrs[i].uri, name, rc);
+			goto err_group;
+		}
+
+		rc = crt_group_psr_set(group, psrs[i].rank);
+		if (rc != 0) {
+			D_ERROR("failed to set rank %u as group %s PSR: %d\n",
+				psrs[i].rank, name, rc);
+			goto err_group;
+		}
+	}
+
+out:
+	*groupp = group;
+	return 0;
+
+err_group:
+	crt_group_view_destroy(group);
+err:
+	return rc;
+}
+
+static void
+detach_group(bool server, bool pmixless, crt_group_t *group)
+{
+	int rc = 0;
+
+	if (!pmixless)
+		rc = crt_group_detach(group);
+	else if (!server)
+		rc = crt_group_view_destroy(group);
+	D_ASSERTF(rc == 0, "%d\n", rc);
+}
+
+static int
+attach(const char *name, int npsrbs, struct psr_buf *psrbs,
+       struct dc_mgmt_sys **sysp)
+{
+	struct dc_mgmt_sys     *sys;
+	crt_group_t	       *group;
+	bool			pmixless = false;
+	int			rc;
+
+	D_DEBUG(DB_MGMT, "attaching to system '%s'\n", name);
+
+	D_ALLOC_PTR(sys);
+	if (sys == NULL) {
+		rc = -DER_NOMEM;
+		goto err;
+	}
+	D_INIT_LIST_HEAD(&sys->sy_link);
+	rc = snprintf(sys->sy_name, sizeof(sys->sy_name), "%s", name);
+	D_ASSERTF(rc >= 0, "%d\n", rc);
+	if (rc >= sizeof(sys->sy_name)) {
+		D_ERROR("system name %s longer than %zu bytes\n", sys->sy_name,
+			sizeof(sys->sy_name) - 1);
+		rc = -DER_OVERFLOW;
+		goto err_sys;
+	}
+
+	group = crt_group_lookup((char *)name);
```

Comment:
is the attach() func to support attach to a remote service group? here the crt_group_lookup() will succeed if pass in name as a local group name or sub-group name. so if pass a local group name to attach() it will sucess. not sure if it will cause confuse?

Reply:
right, it should work as you expected.